test/spec/services/apiServiceSpec.js

```diff
@@ -234,6 +235,96 @@ describe("APIService", function(){
+    var onlyClusterSample = ['ClusterResourceQuota','Namespace','OAuthAccessToken','PersistentVolume','ProjectRequest','User'];
+
+    // at minimum, prove with tests that it is deduplicating by ignoring openshift!
+    it('should return list of kinds that are scoped to a namespace by default', function() {
+      var namespacedKinds = _.map(APIService.availableKinds(), 'kind');
+      // from the fixtures included, we should have 51 namespaced kinds
+      expect( _.difference(bothSample, namespacedKinds).length ).toEqual(0);
+    });
+
+    it('should return list of all kinds, including those that are cluster scoped, when passed a truthy argument', function() {
+      var allKinds = _.map(APIService.availableKinds(true), 'kind');
+      // from the fixtures included, we should have 84 total kinds
+      expect( _.difference(bothSample, allKinds).length ).toEqual(0);
+      expect( _.difference(onlyClusterSample, allKinds).length ).toEqual(0);
+    });
+
+    // kinds from the old /oapi should not be iterated at all.
+    it('should not list kinds from the old /oapi namespace (that do not have a group)', function() {
+      var allKinds = APIService.availableKinds(true);
+      var shouldNotBeFound = [];
+      // This is a sampling of items from /oapi that should no longer be listed
+      var oapiShouldNotExistSample = [
+        {kind: 'ClusterPolicy'},
+        {kind: 'ClusterRole'},
+        {kind: 'Image'},
+        {kind: 'Template'},
+        {kind: 'Project'},
+        {kind: 'User'}
+      ];
+      _.each(oapiShouldNotExistSample, function(kindToFind) {
+        var found = _.find(allKinds, function(kind) {
+          return (kind.kind === kindToFind.kind) && !_.includes(_.keys(kind), 'group');
+        });
+        if(found) {
+          shouldNotBeFound.push(found);
+        }
+      });
+      expect(shouldNotBeFound.length).toEqual(0);
+    });
+
+    // unlike the /oapi endpoint, the /api endpoint should still be listed
+    it('should list items from the k8s /api namespace (that do not have a group)', function() {
+      var allKinds = APIService.availableKinds(true);
+      var shouldBeFound = [];
+      // this is a sampling of items from /api that should still be listed,
+      // even though they do not yet have a group associated.
+      var k8sAPIStillExistsSample = [
+        {kind: 'Binding'},
+        {kind: 'ConfigMap'},
+        {kind: 'Namespace'},
+        {kind: 'PersistentVolume'},
+        {kind: 'Service'},
+        {kind: 'ServiceAccount'},
+        {kind: 'Pod'}
+      ];
+      _.each(k8sAPIStillExistsSample, function(kindToFind) {
+        var found = _.find(allKinds, function(kind) {
+          return (kind.kind === kindToFind.kind) && !_.includes(_.keys(kind), 'group');
+        });
+        if(found) {
+          shouldBeFound.push(found);
+        }
+      });
+      expect(shouldBeFound.length).toEqual(7);
```

Comment:
instead of hardcoding 7 here, the test-case would be more future-proof if you did k8sAPIStillExistsSample.length  that way if someone updates the list they dont need to remember to update the number here